### PR TITLE
Add more search tests.

### PR DESCRIPTION
**Pull request by samwho**
## Description

This adds tests for more variants of search queries against string fields.

The commented add tests fail variously across all datasources, will need to investigate that tomorrow.

### CODE REVIEW

File: packages/server/src/api/routes/tests/search.spec.ts [hunk 1]
```diff
@@ -53,22 +53,41 @@ describe.each([
...
-    ])
+  describe("strings", () => {
+    interface StringSearchTest {
+      query: SearchFilters
+      expected: number[]
+    }
+
+    const stringSearchTests: StringSearchTest[] = [
+      { query: {}, expected: [0, 1] },
+      { query: { string: { name: "foo" } }, expected: [0] },
```
Comment (adrinr) [inline packages/server/src/api/routes/tests/search.spec.ts:64]: This was not really clear on first read. Could we rename it to expectedRowIdx or similar?

**Review by mike12345567 — APPROVED** [review]
Pre-approving this, but agree with @adrinr comment, might be helpful to explain the array of numbers as row indexes.

**Comment by samwho** [comment]
I've gone a slightly different direction, because I didn't really like the indexes at all to be honest. How is it now?

**Review by mike12345567 — APPROVED** [review]
Very nice - was wondering if this would be possible, feels easier to just specify what you expect back!

**Pull request merged**